util/icon/extract_timings.py

```diff
@@ -17,12 +17,34 @@
 sec_regex = r"(\d+[.]?\d*)s"
 number_regex = r"(\d+[.]?\d*)"
 
-date_regex = r"(?:[A-Z][a-z]{2} +){2}\d{1,2} \d{2}:\d{2}:\d{2} [A-Z]{3,4} 20\d{2}"
-dateline_regex = date_regex
+dateline_regex1 = r"(?:[A-Z][a-z]{2} +){2}\d{1,2} \d{2}:\d{2}:\d{2} [A-Z]{3,4} 20\d{2}"
+dateline_regex2 = (
```

Comment:
I think this should be possible in a single regex. But i would prefer anyway to maybe have an array of regex datelines to simplify extending it further in case tasna (or whichever machine we want to use this on) has another timestamp.

Reply:
Yes, I agree it could be done in one regex, but then you would need to try multiple icon_date_format s, to see which one you get. So we will have a set of regex matching one icon_date_format.

Reply:
So, no array of regex to simplify extending and looping over the regexs instead of if else logic for all regexs?

Reply:
I agree, that is nicer.